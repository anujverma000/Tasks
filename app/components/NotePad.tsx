import React, {useState, useRef} from 'react';
import {TextInput, View, StyleSheet} from 'react-native';
import {NotesProps} from '../types';
import Button from './Button';

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#f4f4f6',
    padding: 24,
  },
  headingInput: {
    paddingHorizontal: 12,
    fontSize: 16,
    fontWeight: 'bold',
    backgroundColor: 'white',
    borderTopLeftRadius: 6,
    borderTopRightRadius: 6,
    elevation: 1,
  },
  descriptionInput: {
    paddingHorizontal: 12,
    fontSize: 14,
    lineHeight: 24,
    backgroundColor: 'white',
    borderBottomLeftRadius: 6,
    textAlignVertical: 'top',
    borderBottomRightRadius: 6,
    justifyContent: 'flex-start',
    elevation: 1,
  },
  buttonContainer: {
    marginVertical: 12,
  },
});
interface Props {
  titlePlaceholder: string;
  descriptionPlaceholder: string;
  submitText: string;
  onSubmit: (props: NotesProps) => void;
  title?: string;
  description?: string;
}

const NotePad = (props: Props) => {
  const [title, setTitle] = useState<string>(props.title || '');
  const [description, setDescription] = useState<string>(
    props.description || '',
  );
  const headingEl = useRef<TextInput>(null);
  const descriptionEl = useRef<TextInput>(null);
  const onSave = () => {
    const note: NotesProps = {title, description};
    props.onSubmit(note);
  };

  return (
    <View style={styles.container}>
      <TextInput
        ref={headingEl}
        style={styles.headingInput}
        placeholder={props.titlePlaceholder}
        autoCapitalize="none"
        returnKeyType="next"
        onSubmitEditing={() => descriptionEl.current?.focus()}
        blurOnSubmit={false}
        value={title}
        onChangeText={setTitle}
        placeholderTextColor={'#8C8080'}
        autoFocus={true}
      />
      <TextInput
        ref={descriptionEl}
        style={styles.descriptionInput}
        placeholder={props.descriptionPlaceholder}
        autoCompleteType="off"
        returnKeyType="send"
        autoCapitalize="none"
        value={description}
        multiline={true}
        numberOfLines={6}
        onChangeText={setDescription}
        placeholderTextColor={'#8C8080'}
      />
      <View style={styles.buttonContainer}>
        <Button label={props.submitText} onPress={onSave} />
      </View>
    </View>
  );
};

export default NotePad;
