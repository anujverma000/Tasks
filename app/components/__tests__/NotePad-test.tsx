import React from 'react';
import renderer from 'react-test-renderer';
import NotePad from '../NotePad';

it('NotePad renders correctly', () => {
  const props = {
    titlePlaceholder: 'title Place holder',
    descriptionPlaceholder: 'description Placeholder',
    submitText: 'Save',
    onSubmit: () => {},
  };
  const tree = renderer.create(<NotePad {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
