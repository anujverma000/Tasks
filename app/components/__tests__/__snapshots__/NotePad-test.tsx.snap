// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotePad renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f4f4f6",
      "padding": 24,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoFocus={true}
    blurOnSubmit={false}
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="title Place holder"
    placeholderTextColor="#8C8080"
    rejectResponderTermination={true}
    returnKeyType="next"
    style={
      Object {
        "backgroundColor": "white",
        "borderTopLeftRadius": 6,
        "borderTopRightRadius": 6,
        "elevation": 1,
        "fontSize": 16,
        "fontWeight": "bold",
        "paddingHorizontal": 12,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCompleteType="off"
    multiline={true}
    numberOfLines={6}
    onChangeText={[Function]}
    placeholder="description Placeholder"
    placeholderTextColor="#8C8080"
    rejectResponderTermination={true}
    returnKeyType="send"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "elevation": 1,
        "fontSize": 14,
        "justifyContent": "flex-start",
        "lineHeight": 24,
        "paddingHorizontal": 12,
        "textAlignVertical": "top",
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "marginVertical": 12,
      }
    }
  />
</View>
`;
